src/Passes/Source/Profile/Profile.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Profile/Profile.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    Profile::Profile(
```

Comment:
Nice. I'll need to read the docs in detail to fully understand how all components are structured precisely and where the extensibility points are - but at first glance this looks like a good setup.